### PR TITLE
Use SolveCore API for optimization solvers

### PR DESCRIPTION
**Pull request by geoffroyleconte**

### CODE REVIEW

File: src/RipQP.jl [hunk 1]
```diff
@@ -37,6 +37,42 @@ include("utils.jl")
 
 const to = TimerOutput()
 
+keep_init_prec(mode::Symbol) = (mode == :mono || mode == :ref || mode == :zoom)
+
+include("mono_solver.jl")
+include("two_steps_solver.jl")
+include("three_steps_solver.jl")
+
+function ripqp(
```
Comment (geoffroyleconte) [inline src/RipQP.jl:46]: @tmigot I made these changes following our discussion (there are a lot of modifications but you can look at this short function only).
`RipQPSolver` uses dispatch based on `ap::AbstractRipQPParameters` (there are different parameters structures used to change the settings to solve the problem).
`stats_inner` are the stats of the preprocessed QuadraticModel `QM_inner`, and `get_stats_outer` returns the stats of the initial QuadraticModel `QM`.

**Review by tmigot — APPROVED** [review]
That looks great, thanks!

The motivation behind all this is to later be able to do "in-place"-like restart like that for instance
https://github.com/JuliaSmoothOptimizers/JSOSolvers.jl/blob/main/test/restart.jl .

File: src/RipQP.jl (hunk 1 above)
Comment (tmigot) [reply src/RipQP.jl:46]: In the docstring of `ripqp`, you can also add a block like this (inspired form https://github.com/JuliaSmoothOptimizers/JSOSolvers.jl/blob/main/src/R2.jl#L8 ):
```
For advanced usage, first define a `R2Solver` to preallocate the memory used in the algorithm, and then call `solve!`:
    solver = R2Solver(nlp)
    solve!(solver, nlp; kwargs...)
```

**Comment by geoffroyleconte** [comment]
> That looks great, thanks!
> 
> The motivation behind all this is to later be able to do "in-place"-like restart like that for instance https://github.com/JuliaSmoothOptimizers/JSOSolvers.jl/blob/main/test/restart.jl .

Thanks for the review! So can I use the solution/multipliers from the preallocated stats for the in-place restart? Or do I have to use the data from the `NLPModelMeta` `meta.x0, meta.y0` (which does not have preallocated bound multipliers)?

**Comment by tmigot** [comment]
Good question :). For `x0` and `y0`, I would say we need to use the `NLPModelMeta`, and for the bounds it is maybe in the `RipQPSolver` ? For now, as long as, this would be documented I think that's great in my opinion.

**Pull request merged**